lib/install/management/appliance.go

```diff
@@ -149,6 +149,20 @@ func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine, force bool) error {
 	defer trace.End(trace.Begin(fmt.Sprintf("vm %q, force %t", vm.String(), force)))
 
 	var err error
+	// get the actual folder name before we delete it
+	folder, err := vm.FolderName(d.ctx)
+	if err != nil {
+		// failed to get folder name, might not be able to remove files for this VM
+		name := d.getName(vm)
+		if name == "" {
+			log.Errorf("Unable to automatically remove all files in datastore for VM %q", vm.Reference())
+		} else {
+			// try to use the vm name in place of folder
+			log.Infof("Delete will attempt to remove datastore files for VM %q", name)
```

Comment:
I think this log message needs a better phrasing. However, I am not the best person to suggest a better text here. @matthewavery ?